lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -185,28 +185,31 @@ def package_manager
       end
 
       def os_facts
-        facter = ForemanMaintain::Utils::Facter.path
-        @os_facts ||= JSON.parse(execute("#{facter} -j os"))
+        @os_facts ||= ForemanMaintain::Utils::OsFacts
```

Comment:
`OsFacts` is a class. I'm not sure if there is a benefit to keeping `@os_facts`
```suggestion
        ForemanMaintain::Utils::OsFacts
```